src/General/Tag/Tag.tsx

```diff
@@ -1,38 +1,65 @@
 import * as React from 'react';
 import classNames from 'classnames';
-import { TagContainer, TagContent } from '../../Style/General/TagStyle';
+import {
+  TagContainer,
+  TagContent,
+  StartIconContainer,
+  EndIconContainer,
+} from '../../Style/General/TagStyle';
 
 const Tag: React.FunctionComponent<Props> = props => {
-  const { className, children, theme, block, outline, ...restProps } = props;
+  const {
+    className,
+    children,
+    block,
+    outline,
+    border,
+    startIcon,
+    endIcon,
+    onClick,
+    ...restProps
+  } = props;
+
+  const content = (
+    <>
+      {startIcon && <StartIconContainer>{startIcon}</StartIconContainer>}
+      {children}
+      {endIcon && <EndIconContainer>{endIcon}</EndIconContainer>}
+    </>
+  );
 
   return (
     <React.Fragment>
       <TagContainer
         className={classNames('aries-tag', className)}
         role="presentation"
         tabIndex={0}
-        theme={theme}
         block={block}
         outline={outline}
+        border={border}
+        isClickable={typeof onClick === 'function'}
+        onClick={onClick}
         {...restProps}
       >
         <TagContent className="tag-content" tabIndex={-1}>
-          {children}
+          {content}
         </TagContent>
       </TagContainer>
     </React.Fragment>
   );
 };
 
 Tag.defaultProps = {
-  theme: 'grey',
   block: false,
   outline: false,
+  border: 'solid',
 };
 
 interface Props extends React.ComponentPropsWithoutRef<typeof TagContainer> {
   children: React.ReactNode;
-  theme?: string;
+  onClick?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
+  startIcon?: React.ReactNode;
+  endIcon?: React.ReactNode;
```

Comment:
shall we let users pass any startIcon  and endIcon? It seems like only 3 icons are in the design system
![Screen Shot 2020-05-06 at 1 53 58 PM](https://user-images.githubusercontent.com/11376798/81142057-0e430000-8fa1-11ea-9f02-c4848939608f.png)

Reply:
same as https://github.com/glints-dev/glints-aries/pull/323#issuecomment-624464171